controllers/chat/initChatFromPost.js

```diff
@@ -129,6 +129,19 @@ const initChatFromPost = async (req, res) => {
           };
         }
 
+        if (targetUserModel.is_anonymous && !userModel.is_anonymous) {
+          anonDetailInfo = {
+            anon_user_info_color_code:
+              newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_color_code,
+            anon_user_info_color_name:
+              newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_color_name,
+            anon_user_info_emoji_code:
+              newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_emoji_code,
+            anon_user_info_emoji_name:
+              newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_emoji_name
+          };
+        }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code assumes that `newStateMemberWithAnonInfo[targetUserModel.user_id]` will always exist and have the necessary properties. If it doesn't, this could lead to a TypeError. It would be safer to check if these properties exist before trying to access them.

```diff
-          anonDetailInfo = {
-            anon_user_info_color_code:
-              newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_color_code,
-            anon_user_info_color_name:
-              newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_color_name,
-            anon_user_info_emoji_code:
-              newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_emoji_code,
-            anon_user_info_emoji_name:
-              newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_emoji_name
-          };
+          const targetAnonInfo = newStateMemberWithAnonInfo[targetUserModel.user_id];
+          if (targetAnonInfo) {
+            anonDetailInfo = {
+              anon_user_info_color_code: targetAnonInfo.anon_user_info_color_code,
+              anon_user_info_color_name: targetAnonInfo.anon_user_info_color_name,
+              anon_user_info_emoji_code: targetAnonInfo.anon_user_info_emoji_code,
+              anon_user_info_emoji_name: targetAnonInfo.anon_user_info_emoji_name
+            };
+          }
```
This way, we only try to access the properties if `targetAnonInfo` is not undefined or null.


<!-- This is an auto-generated comment by OSS CodeRabbit -->